src/NServiceBus.NHibernate/SharedSession/SharedSessionConfig.cs

```diff
@@ -0,0 +1,21 @@
+namespace NServiceBus.Persistence.NHibernate
+{
+    using Configuration.AdvanceExtensibility;
+    using NServiceBus.Features;
+
+    /// <summary>
+    /// Deduplication configuration extensions.
+    /// </summary>
+    public static class SharedSessionConfig
+    {
+        /// <summary>
+        /// Disables sharing a connection between SQL Server transport and NHibernate persistence. Intended to be used in conjunction with Outbox only.
+        /// </summary>
+        /// <param name="persistenceConfiguration"></param>
+        public static PersistenceExtentions<NHibernatePersistence> DoNotShareTransportConnection(this PersistenceExtentions<NHibernatePersistence> persistenceConfiguration)
```

Comment:
> The void would break the chaining of the fluent API.

My point exactly. Fluent APIs should have terminating methods. Chaining something off `DoNotShareTransportConnection` implies you are mutating the result of `DoNotShareTransportConnection` and that order is important, neither of which is correct